MetasysServices/Trends/TrendServiceProvider.cs

```diff
@@ -36,30 +41,36 @@ public PagedResult<Sample> GetSamples(Guid objectId, int attributeId, TimeFilter
         /// <inheritdoc/>
         public async Task<PagedResult<Sample>> GetSamplesAsync(Guid objectId, int attributeId, TimeFilter filter)
         {
-            List<Sample> objectSamples = new List<Sample>();                   
+            List<Sample> objectSamples = new List<Sample>();
             // Perform a generic call using objects resource valid for Network Devices as well
             var response = await GetPagedResultsAsync<JToken>("objects", ToDictionary(filter), objectId, "attributes", attributeId,"samples").ConfigureAwait(false);
             // Read full attribute from url
             foreach (JToken s in response.Items)
             {
                 Sample sample = new Sample();
-                string unitsUrl=string.Empty;
+                string unitsUrl=string.Empty;
+                string UnitEnumerationKey = string.Empty;
+                string UnitKey = "unitEnumSet.degC";
```

Comment:
What is the logic behind this assumption? UnitKey could be anything then DegC, the solution could be to read the unit description from UnitsUrl (doing appropriate caching) and then set the Unit property with that value. If it is not possible to get the UnitKey in v2, then would be just null, however you could try to retrieve the enumeration from the description of unitUrl using the GetObjectTypeEnumeration method or something similar that does reverse engineering on the dictionary.